infra/modules/providers/azure/data-factory/README.md

```diff
@@ -0,0 +1,130 @@
+
+For more information, Please check Microsoft Azure Data Factory [Documentation](https://docs.microsoft.com/en-us/azure/data-factory/introduction).
+
+## Characteristics
+
+An instance of the `data-factory` module deploys the _**Data Factory**_ in order to provide templates with the following:
+
+- Ability to provision a single Data Factory instance
+- Ability to provision a configurable Pipeline
+- Ability to configure Trigger
+- Ability to configure SQL server Dataset
+- Ability to configure SQL server Linked Service
+
+## Out Of Scope
+
+The following are not support in the time being
+
+- Creating Multiple pipelines
+- Only SQL server Dataset/Linked Service are implemented.
+
+## Definition
+
+Terraform resources used to define the `data-factory` module include the following:
+
+- [azurerm_data_factory](https://www.terraform.io/docs/providers/azurerm/r/data_factory.html)
+- [azurerm_data_factory_integration_runtime_managed](https://www.terraform.io/docs/providers/azurerm/r/data_factory_integration_runtime_managed.html)
+- [azurerm_data_factory_pipeline](https://www.terraform.io/docs/providers/azurerm/r/data_factory_pipeline.html)
+- [azurerm_data_factory_trigger_schedule](https://www.terraform.io/docs/providers/azurerm/r/data_factory_trigger_schedule.html)
+- [azurerm_data_factory_dataset_sql_server](https://www.terraform.io/docs/providers/azurerm/r/data_factory_dataset_sql_server_table.html)
+- [azurerm_data_factory_linked_service_sql_server](https://www.terraform.io/docs/providers/azurerm/r/data_factory_linked_service_sql_server.html)
+
+## Usage
+
+Data Factory usage example:
+
+```terraform
+module "data_factory" {
+    source              = "../../modules/providers/azure/data-factory"
+    data_factory_name      = "adf"
+    resource_group_name = "rg"
+    data_factory_runtime_name                 = "adfrt"
+    node_size                = "Standard_D2_v3"
+    number_of_nodes = 1
+    edition = "Standard"
+    max_parallel_executions_per_node = 1
+    vnet_integration = {
+                    vnet_id     = "/subscriptions/resourceGroups/providers/Microsoft.Network/virtualNetworks/testvnet"
+                    subnet_name = "default"
+                    }
+    data_factory_pipeline_name = "adfpipeline"
+    data_factory_trigger_name = "adftrigger"
+    data_factory_trigger_interval = 1
+    data_factory_trigger_frequency = "Minute"
+    data_factory_dataset_sql_name = "adfsqldataset"
+    data_factory_dataset_sql_table_name = "adfsqldatasettable"
+    data_factory_dataset_sql_folder = ""
+    data_factory_linked_sql_name = "adfsqllinked"
+    data_factory_linked_sql_connection_string = "Server=tcp:adfsql..."
+}
+```
+
+## Outputs
+
+The value will have the following schema:
+
+```terraform
```

Comment:
[nit] referencing the output file (like you did w/ vars file) will make this document more robust over time as the module changes.

Reply:
removed the output block and referenced the file instead